runtime/src/lib.rs

```diff
@@ -24,733 +24,40 @@
-    SignedAccountId32AsNative<NodleNetwork, Origin>,
-);
-
-parameter_types! {
-    pub NativeReserveTokens: BTreeSet<(Vec<u8>, MultiLocation)> = {
-        let mut t = BTreeSet::new();
-        //TODO: might need to add other assets based on orml-tokens
-        t.insert(("AUSD".into(), (Junction::Parent, Junction::Parachain { id: 666 }).into()));
-        t
-    };
-}
-
-pub struct XcmConfig;
-impl Config for XcmConfig {
-    type Call = Call;
-    type XcmSender = XcmHandler;
-    type AssetTransactor = LocalAssetTransactor;
-    type OriginConverter = LocalOriginConverter;
-    type IsReserve = NativePalletAssetOr<NativeReserveTokens>;
-    type IsTeleporter = ();
-    type LocationInverter = LocationInverter<Ancestry>;
-}
-
-impl cumulus_pallet_xcm_handler::Config for Runtime {
-    type Event = Event;
-    type XcmExecutor = XcmExecutor<XcmConfig>;
-    type UpwardMessageSender = ParachainSystem;
-    type HrmpMessageSender = ParachainSystem;
-    type SendXcmOrigin = EnsureRoot<AccountId>;
-    type AccountIdConverter = LocationConverter;
-}
-
-pub struct RelayToNative;
-impl Convert<RelayChainBalance, Balance> for RelayToNative {
-    fn convert(val: u128) -> Balance {
-        // both native and relay have 12 decimals
-        val
-    }
-}
-
-pub struct NativeToRelay;
-impl Convert<Balance, RelayChainBalance> for NativeToRelay {
-    fn convert(val: u128) -> Balance {
-        // both native and relay have 12 decimals
-        val
-    }
-}
-
-pub struct HandleXcm;
-impl XcmHandlerT<AccountId> for HandleXcm {
-    fn execute_xcm(origin: AccountId, xcm: Xcm) -> DispatchResult {
-        XcmHandler::execute_xcm(origin, xcm)
-    }
-}
-
-impl orml_xtokens::Config for Runtime {
-    type Event = Event;
-    type Balance = Balance;
-    type ToRelayChainBalance = NativeToRelay;
-    type AccountId32Convert = AccountId32Convert;
-    type RelayChainNetworkId = PolkadotNetworkId;
-    type ParaId = ParachainInfo;
-    type XcmHandler = HandleXcm;
-}
-
-parameter_type_with_key! {
-    pub ExistentialDeposits: |_currency_id: CurrencyId| -> Balance {
-        Zero::zero()
-    };
-}
-
-impl orml_tokens::Config for Runtime {
-    type Event = Event;
-    type Balance = Balance;
-    type Amount = Amount;
-    type CurrencyId = CurrencyId;
-    type WeightInfo = ();
-    type ExistentialDeposits = ExistentialDeposits;
-    type OnDust = orml_tokens::TransferDust<Runtime, CompanyReserveAccount>;
-}
-
-parameter_types! {
-    pub const GetNodleTokenId: CurrencyId = CurrencyId::NODL;
-}
-
-pub type NodleToken = BasicCurrencyAdapter<Runtime, Balances, Amount, BlockNumber>;
-
-impl orml_currencies::Config for Runtime {
-    type Event = Event;
-    type MultiCurrency = orml_tokens::Pallet<Runtime>;
-    type NativeCurrency = NodleToken;
-    type GetNativeCurrencyId = GetNodleTokenId;
-    type WeightInfo = ();
-}
-
-impl orml_unknown_tokens::Config for Runtime {
-    type Event = Event;
-}
-
 construct_runtime!(
```

Comment:
Are we not merging the changes with the custom macros?

Reply:
You are right, right now it's more like stubbed or disabled with partial implementation of '#[cfg(feature = "enable-contracts")]', Idea is to remove the annotation once package dependency is resolved.

And we can't use CLI args like `--features enable-contracts` to control the build.